src/main/java/io/github/thebusybiscuit/slimefun4/implementation/listeners/AutoCrafterListener.java

```diff
@@ -58,6 +65,21 @@ public void onInteract(PlayerRightClickEvent e) {
                     return;
                 }
 
+                // Check if the recipe of the item is disabled.
+                boolean unlocked = true;
+                for (Recipe recipe : Bukkit.getRecipesFor(e.getItem())) {
+                    if (!e.getPlayer().hasDiscoveredRecipe(((Keyed) recipe).getKey())) {
+                        unlocked = false;
+                        break;
+                    }
+                }
+
+                if (e.getPlayer().getWorld().getGameRuleValue(GameRule.DO_LIMITED_CRAFTING) && !unlocked) {
```

Comment:
When was this gamerule added...? 
We will likely need a version check here.

Reply:
It was added in 17w13a which is in 1.12. So I think it would be fine... right?

Reply:
Alright then. I literally only heard about that gamerule through this pull request, so I assumed it was new XD
Guess I lived under a rock.